### PR TITLE
[`Mixtral`] Add more details on quantization

### PR DESCRIPTION
**Pull request by younesbelkada**
# What does this PR do?

This PR introduces a new section in the Mixtral blogpost by providing details on the quantization support on HF ecosystem with that model, including GPTQ and QLoRA. I will update it once we have AWQ support as well

cc @osanseviero @SunMarc

### CODE REVIEW

File: mixtral.md [hunk 1]
```diff
@@ -216,6 +216,75 @@ accelerate launch --config_file examples/accelerate_configs/multi_gpu.yaml --num
 
 This takes about 48 hours to train on a single A100, but can be easily parallelised by tweaking `--num_processes` to the number of GPUs you have available.
 
+## Quantizing Mixtral-7B
+
+As seen above, the challenge for this model is to make it run on consumer-type hardware for anyone to use it, as the model requires ~90GB just to be loaded in half-precision (`torch.float16`).
+
+With 🤗 transformers library we support out of the box inference with state-of-the-art quantization methods such as QLoRA, and GPTQ. You can read more about the quantization methods we support in the [appropriate documentation section](https://huggingface.co/docs/transformers/quantization).
```
Comment (osanseviero) [inline mixtral.md:223]:
```suggestion
With the 🤗 transformers library, we support out-of-the-box inference with state-of-the-art quantization methods such as QLoRA and GPTQ. You can read more about the quantization methods we support in the [appropriate documentation section](https://huggingface.co/docs/transformers/quantization). 
```

File: mixtral.md [hunk 2]
```diff
@@ -216,6 +216,75 @@ accelerate launch --config_file examples/accelerate_configs/multi_gpu.yaml --num
...
 
+## Quantizing Mixtral-7B
+
+As seen above, the challenge for this model is to make it run on consumer-type hardware for anyone to use it, as the model requires ~90GB just to be loaded in half-precision (`torch.float16`).
+
+With 🤗 transformers library we support out of the box inference with state-of-the-art quantization methods such as QLoRA, and GPTQ. You can read more about the quantization methods we support in the [appropriate documentation section](https://huggingface.co/docs/transformers/quantization). 
+
+### Load Mixtral with QLoRA
+
+As demonstrated in the inference section, you can load Mixtral with QLoRA by installing `bitsandbytes` library `pip install -U bitsandbytes` and passing the flag `load_in_4bit=True` to from_pretrained method. For better performance, we advise users to load the model with `bnb_4bit_compute_dtype=torch.float16`. Note you need a GPU device with at least 24GB VRAM in order to properly run the snippet below.
```
Comment (osanseviero) [inline mixtral.md:227]:
```suggestion
As demonstrated in the inference section, you can load Mixtral with 4-bit quantization by installing `bitsandbytes` library `pip install -U bitsandbytes` and passing the flag `load_in_4bit=True` to the `from_pretrained` method. For better performance, we advise users to load the model with `bnb_4bit_compute_dtype=torch.float16`. Note you need a GPU device with at least 24GB VRAM to properly run the snippet below.
```

**Review by SunMarc — APPROVED** [review]
LGTM ! Thanks for adding this

File: mixtral.md [hunk 3]
```diff
@@ -216,6 +216,75 @@ accelerate launch --config_file examples/accelerate_configs/multi_gpu.yaml --num
...
+model_id = "mistralai/Mixtral-8x7B-Instruct-v0.1"
+tokenizer = AutoTokenizer.from_pretrained(model_id)
+
+quantization_config = BitsAndBytesConfig(
+    load_in_4bit=True,
+    bnb_4bit_compute_dtype=torch.float16
+)
+model = AutoModelForCausalLM.from_pretrained(model_id, quantization_config=quantization_config)
+
+prompt = "USER: Explain what a Mixture of Experts is in less than 100 words. ASSISTANT:"
```
Comment (osanseviero) [inline mixtral.md:242]: Maybe use an example with correct prompt structure

File: mixtral.md (hunk 2 above)
Comment (osanseviero) [inline mixtral.md:227]: Here you say 24GB VRAM but below 30

**Review by osanseviero — APPROVED** [review]
Very cool :fire:

File: mixtral.md (hunk 3 above)
Comment (younesbelkada) [reply mixtral.md]: Nice catch ! changed it!

File: mixtral.md (hunk 2 above)
Comment (younesbelkada) [reply mixtral.md]: Ah yes true, changed it as well

**Pull request merged**